analyses/cell-type-ewings/template_notebooks/auc-singler-workflow/03-clustering.Rmd

```diff
@@ -0,0 +1,265 @@
+
+Now we will compare the clustering results to the cell type assignments obtained from `SingleR` in the `aucell-singler-annotation.sh` workflow. 
+To compare results we will calculate the Jaccard similarity index between clusters and cell types. 
+We expect that good clustering will line up with the cell type assignments so that there is ~ 1 cell type per cluster. 
+
+For the plots, we will only display the top 7 cell types identified by `SingleR` and all other cells will be lumped together into `All remaining cell types`. 
+
+```{r}
+cluster_classification_df <- all_cluster_results |>
+  # add in classifications from singler
+  dplyr::left_join(singler_classification_df, by = "barcodes") |>
+  dplyr::mutate(
+    # first grab anything that is tumor and label it tumor
+    # NA should be unknown
+    singler_annotation = dplyr::case_when(
+      stringr::str_detect(singler_annotation, "tumor") ~ "tumor",
+      is.na(singler_annotation) ~ "unknown", # make sure to separate out unknown labels
+      .default = singler_annotation
+    ) |>
+      forcats::fct_relevel("tumor", after = 0),
+    # get the top cell types for plotting later
+    singler_lumped = singler_annotation |>
+      forcats::fct_lump_n(7, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf),
+    nn_param = forcats::fct_relevel(nn_param, "k_5", after = 0)
+  )
+```
+
+```{r, fig.height=10, fig.width=7}
+# get heatmap for showing cluster assignments vs. singler assignments
+cluster_celltype_heatmap(cluster_classification_df)
+```
+
+
+## Marker gene expression across clusters 
+
+Finally, we will look at the expression of marker genes across each cluster for each value of K. 
+In these plots, each row shows the distribution of the specified marker genes in that cluster. 
+Each panel is labeled with the marker gene expression being plotted. 
+
+```{r}
+# read in marker genes table
+marker_genes_df <- readr::read_tsv(params$marker_genes_file, show_col_types = FALSE) |>
+  # account for genes being from multiple sources
+  dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |>
+  dplyr::distinct()
+
+# get list of all cell types found
+cell_types <- unique(marker_genes_df$cell_type)
+
+# get the sum of expression of all genes for each cell type
+gene_exp_df <- cell_types |>
+  purrr::map(\(type){
+    calculate_sum_markers(marker_genes_df, sce, type)
+  }) |>
+  purrr::reduce(dplyr::inner_join, by = "barcodes")
+
+# join sum expression columns with clustering results
+cluster_exp_df <- all_cluster_results |>
+  dplyr::left_join(gene_exp_df, by = "barcodes")
+```
+
+```{r, fig.height=10, fig.width=5}
+# plot to look at marker gene expression for all cell types across all clusters and all params
+k_params <- unique(cluster_exp_df$nn_param) |>
+  as.character()
+
+k_params |>
+  purrr::map(\(k) {
+    plot_marker_genes(cluster_exp_df,
```

Comment:
Looking at this again in this template, I think we should wrap these plots, and/or maybe also only look at not-uncommon cell types (maybe the top 10 or so, and have a 5x2 grid?). Otherwise this section could get really long and out of hand fast!

Reply:
Just an FYI that this will only ever look at tumor, immune, endothelial, and mesenchymal like cells. So there will always just be 4 density plots per value of k.